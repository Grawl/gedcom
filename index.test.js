const fs = require("fs");
const { tokenize, entities } = require("./");

test("parser", () => {
  expect(
    entities(tokenize(fs.readFileSync("./sample-norefs.ged", "utf8")))
  ).toMatchSnapshot();
});

test("parser-refs", () => {
  expect([...tokenize("1 NAME Elizabeth /Stansfield/")]).toMatchSnapshot();
});
