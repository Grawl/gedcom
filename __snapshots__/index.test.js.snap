// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser 1`] = `
Object {
  "all": Map {},
  "families": Map {},
  "people": Map {},
}
`;

exports[`parser-refs 1`] = `
Array [
  Object {
    "raw": "1",
    "type": "LEVEL",
    "value": 1,
  },
  Object {
    "raw": " ",
    "type": "DELIM",
  },
  Object {
    "raw": "NAME",
    "type": "TAG",
  },
  Object {
    "raw": "Elizabeth /Stansfield/",
    "type": "LINEVALUE",
  },
]
`;
